docs: add `run android --variant --app-id` for custom Android product flavors (#31785)
expo/expo: docs/pages/more/expo-cli.mdx

<Terminal cmd={['$ open -a /Applications/Android Studio.app android']} />

> If you have a customized Android project using different product flavors, you can configure both the flavor and application id using the `--variant` and `--app-id` flags. Learn more in the [local app development](/guides/local-app-development/#local-builds-using-android-product-flavors) guide:
> 
> <Terminal cmd={['$ npx expo run:android --variant freeDebug --app-id dev.expo.myapp.free']} />

Reviewer: ```suggestion
If you have a customized Android project using different product flavors, you can configure both the flavor and application ID using the `--variant` and `--app-id` flags:

<Terminal cmd={['$ npx expo run:android --variant freeDebug --app-id dev.expo.myapp.free']} />

For more information, see the [Local builds using Android product flavors](/guides/local-app-development/#local-builds-using-android-product-flavors) guide.
```

Let's use the section title instead of guide name when referring the link. Also, adding the link to the guide before the command breaks the flow of text. Let's add it after we've mentioned the command
